### ISSUE TITLE
Is there any solution for COUNT(DISTINCT, column) ?

### ISSUE THREAD

**Issue by pjongy** [tortoise/tortoise-orm#296]
**Is your feature request related to a problem? Please describe.**
I've feel something strange while using tor 'tortoise.functions.Count' that does inherited Aggregate.
For my expectation, Count should be inherit can represent with distinction (ex. DistinctFunction)

For instance, pypika that tortoise is using for query genrerating's Count inherits 'DistinctOptionFunction' and we can use it with distinct builder like this:
```
fn.Count('*').distinct()
```

but tortoise does not admit distinct builder because of structure that follows below:
tortoise.function.Function
```
function_field = self.database_func(field, *default_values)
return {"joins": function_joins, "field": function_field}
```

**Describe the solution you'd like**
Count function can be used with distinct flag like this:
```
class DistincFunction:
    def __init__(..., is_distinct):
    ...
    if is_distinct:
        function_field = self.database_func(field, *default_values).distinct()
...
class Count(DistinctFunction):
    database_func = functions.Count
```

**Describe alternatives you've considered**
.

**Additional context**
.

**Comment by pjongy** [comment]
Now, I made some manual code and using distincted count in my project temporary like this:

```
from pypika import functions
from tortoise import ConfigurationError
from tortoise.functions import Function

class DistinctFunction(Function):
    def __init__(self, field, is_distinct, *default_values) -> None:
        super().__init__(field, *default_values)
        self._is_distinct = is_distinct

    def _resolve_field_for_model(self, model, field: str, *default_values) -> dict:
        field_split = field.split("__")
        if not field_split[1:]:
            function_joins = []
            if field_split[0] in model._meta.fetch_fields:
                related_field = model._meta.fields_map[field_split[0]]
                related_field_meta = related_field.model_class._meta
                join = (model._meta.basetable, field_split[0], related_field)
                function_joins.append(join)
                field = related_field_meta.basetable[related_field_meta.db_pk_field]
            else:
                field = model._meta.basetable[field_split[0]]

                if self.populate_field_object:
                    self.field_object = model._meta.fields_map.get(field_split[0], None)
                    if self.field_object:
                        func = self.field_object.get_for_dialect(
                            model._meta.db.capabilities.dialect, "function_cast"
                        )
                        if func:
                            field = func(self.field_object, field)
            if self._is_distinct:
                function_field = self.database_func(field, *default_values).distinct()
            else:
                function_field = self.database_func(field, *default_values)
            return {"joins": function_joins, "field": function_field}

        if field_split[0] not in model._meta.fetch_fields:
            raise ConfigurationError(f"{field} not resolvable")
        related_field = model._meta.fields_map[field_split[0]]
        join = (model._meta.basetable, field_split[0], related_field)
        function = self._resolve_field_for_model(
            related_field.model_class, "__".join(field_split[1:]), *default_values
        )
        function["joins"].append(join)
        return function


class DistinctCount(DistinctFunction):
    database_func = functions.Count
```

**Comment by grigi** [comment]
Yes, I agree that this is something that we probably need. Thank you for the sample code. :+1:

**Comment by lntuition** [comment]
Fixed by #312, Applied to v0.16.0
